crates/xline/src/conflict/mod.rs

```diff
@@ -0,0 +1,51 @@
+use curp::cmd::Command as CurpCommand;
+use utils::interval_map::Interval;
+use xlineapi::{command::Command, interval::BytesAffine, RequestBackend, RequestWrapper};
+
+/// Speculative pool implementations
+pub(crate) mod spec_pool;
+/// Uncommitted pool implementations
+pub(crate) mod uncommitted_pool;
+
+/// Returns command intervals
+fn intervals<C>(entry: &C) -> Vec<Interval<BytesAffine>>
+where
+    C: AsRef<Command>,
+{
+    entry
+        .as_ref()
+        .keys()
+        .iter()
+        .cloned()
+        .map(Into::into)
+        .collect()
+}
+
+/// Filter kv commands
+fn filter_kv<C>(entry: C) -> Option<C>
+where
+    C: AsRef<Command>,
+{
+    matches!(entry.as_ref().request().backend(), RequestBackend::Kv).then_some(entry)
+}
+
+/// Returns `true` if this command conflicts with all other commands
+fn is_xor_cmd(cmd: &Command) -> bool {
```

Comment:
What does `xor` mean in the function name? It usually means an bit-level operation.

Reply:
A xor command means that the command conflicts with all other commands, the word "xor" implies that the commands are mutually exclusive. Maybe you can propose a better name for this type of commands.

Reply:
\"exclusive_cmd\"?

Reply:
\"exclusive_cmd\"?

Reply:
\"exclusive_cmd\"?

Reply:
\"exclusive_cmd\"?

Reply:
\"exclusive_cmd\"?

Reply:
done